tests/cecil-tests/ApiCapitalizationTest.cs

```diff
@@ -0,0 +1,252 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Versioning;
+using System.Text;
+using System.Text.RegularExpressions;
+using NUnit.Framework;
+
+using Mono.Cecil;
+using Xamarin.Tests;
+using Xamarin.Utils;
+using System.Configuration.Assemblies;
+using MethodAttributes = Mono.Cecil.MethodAttributes;
+
+#nullable enable
+
+namespace Cecil.Tests {
+	[TestFixture]
+	public class ApiCapitalizationTest {
+
+		Dictionary<string, AssemblyDefinition> assemblyCache = new Dictionary<string, AssemblyDefinition> ();
+
+
+		[OneTimeSetUp]
+		public void SetUp ()
+		{
+			foreach (string assemblyPath in Helper.NetPlatformAssemblies) {
+
+				var assembly = Helper.GetAssembly (assemblyPath);
+				assemblyCache.Add (assemblyPath, assembly);
+			}
+		}
+
+		[OneTimeTearDown]
+		public void TearDown ()
+		{
+			assemblyCache.Clear ();
+		}
+
+		bool IsTypeObsolete (TypeDefinition type)
+		{
+			if (type == null)
+				return false;
+
+			if (type.HasCustomAttributes && type.CustomAttributes.Where ((t) => t.AttributeType.Name == "ObsoleteAttribute" || t.AttributeType.Name == "AdviceAttribute").Any ()) {
+				return true;
+			}
+
+			return false;
+		}
+
+
+		bool IsMethodObsolete (MethodDefinition method)
+		{
+			if (method == null)
+				return false;
+
+			if (method.HasCustomAttributes && method.CustomAttributes.Where ((m) => m.AttributeType.Name == "ObsoleteAttribute").Any ()) {
+				return true;
+			}
+
+
+			return false;
+		}
+
+		bool IsPropertyObsolete (PropertyDefinition property, TypeDefinition t)
+		{
+			if (property == null)
+				return false;
+
+			if (property.HasCustomAttributes && property.CustomAttributes.Where ((p) => p.AttributeType.Name == "ObsoleteAttribute" || p.AttributeType.Name == "ObsoletedOSPlatformAttribute").Any ()) {
+				return true;
+			}
+
+			return false;
+		}
+
+
+		bool IsException (string assemblyPath, MethodDefinition m)
```

Comment:
The word `Exception` is already used in a somewhat different meaning in C#, so using a different name might be better to avoid confusion for other people (maybe something like `IsSpecial`, or `SkipMethod`, there are a lot of possibilities! I'm very bad at naming, so feel free to come up with your own alternative)